Initial work on the repository

vpodzime opened the pull request.

vpodzime committed:
chore: Add README documenting the purpose of this repo

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime committed:
chore: Add the docker-compose module

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime committed:
chore: Split the module into docker-compose specific and common parts

So that the common part can be reused by other container Update Modules.

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime committed:
ci: Add elementary CI checks

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime committed:
chore: Use our common LICENSE file

Adding a copyright statement and dropping the instructions
appendix.

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

elkoniu commented on src/container-modules-common.sh:
```diff
@@ -0,0 +1,378 @@
+#!/bin/sh
```
Some good Bash meat:) I am just wondering - if this should not be split in more than one file?

vpodzime replied:
I was thinking about that too, but it kind of doesn't make much sense if it's going to be all or none every time. It's not too long, it just looks long because it's shell "code".

elkoniu replied:
Agree :)

elkoniu commented:
In the close future we will be trying to unify our repos readme and makefilse. Based on current repos state this is a template I crafted so far (and maybe will be usefull for this repo too):
```
![Mender logo](https://github.com/mendersoftware/mender/blob/master/mender_logo.png)

[![Build Status](https://gitlab.com/Northern.tech/Mender/mender/badges/master/pipeline.svg)](https://gitlab.com/Northern.tech/Mender/mender/pipelines)
[![Coverage Status](https://coveralls.io/repos/github/mendersoftware/mender/badge.svg?branch=master)](https://coveralls.io/github/mendersoftware/mender?branch=master)

# Overview

XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX


## Getting started

To start using Mender, we recommend that you begin with the Getting started
section in [the Mender documentation](https://docs.mender.io/).


## Building

XXXXXXXXXXXXXXXXXXXXXXXXXXXXX


## Downloading the binaries

For getting the latest builds of our packages please check documentation [download page](https://docs.mender.io/downloads)


## Contributing

We welcome and ask for your contribution. If you would like to contribute to Mender, please read our guide on how to best get started [contributing code or
documentation](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md).


## License

Mender is licensed under the Apache License, Version 2.0. See
[LICENSE](https://github.com/mendersoftware/artifacts/blob/master/LICENSE) for the
full license text.


## Security disclosure

We take security very seriously. If you come across any issue regarding
security, please disclose the information by sending an email to
[security@mender.io](security@mender.io). Please do not create a new public
issue. We thank you in advance for your cooperation.


## Connect with us

* Join the [Mender Hub discussion forum](https://hub.mender.io)
* Follow us on [Twitter](https://twitter.com/mender_io). Please
  feel free to tweet us questions.
* Fork us on [Github](https://github.com/mendersoftware)
* Create an issue in the [bugtracker](https://northerntech.atlassian.net/projects/MEN)
* Email us at [contact@mender.io](mailto:contact@mender.io)
* Connect to the [#mender IRC channel on Libera](https://web.libera.chat/?#mender)
```

vpodzime committed:
chore: Drop image delta support

To be reworked in MEN-7623.

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime committed:
chore: Simplify container_image_load()

It doesn't need most of its arguments. The current implementation
doesn't even need the container's URL ("global identifier"), but
this may be required in the future or for other container
runtimes.

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime committed:
chore: Rename cleanup() to persistent_store_cleanup()

Because that's what it does and cannot be easily confused with
the `clean_up()` function.

Ticket: MEN-8595
Changelog: none
Signed-off-by: Vratislav Podzimek <vratislav.podzimek+auto-signed@northern.tech>

vpodzime commented:
We don't want to follow this path of extracting bits from the `app-update-module`. Instead, we will create a new `docker-compose` Update Module from scratch, possibly using bits from this PR.
